lib/format.js

```diff
@@ -16,46 +16,41 @@ let format = {
   },
 
   relativeTime: function(timestamp, currentTime = Date.now()) {
-    let msPerMinute = 60 * 1000;
-    let msPerHour = msPerMinute * 60;
-    let msPerDay = msPerHour * 24;
-    let msPerMonth = msPerDay * 30;
-    let msPerYear = msPerDay * 365;
-
-    let difference = currentTime - timestamp;
-
-    let isPlural = (value) => { return ( value==1 ? '' : 's' ); };
-
-    if (difference < msPerMinute) {
-      let i = Math.floor(difference/1000);
-      return i + ' second' + isPlural(i) + ' ago'
-    }
-
-    else if (difference < msPerHour) {
-      let i = Math.floor(difference/msPerMinute);
-      return i + ' minute' + isPlural(i) + ' ago';
-    }
-
-    else if (difference < msPerDay ) {
-      let i = Math.floor(difference/msPerHour);
-      return i + ' hour' + isPlural(i) + ' ago';
-    }
-
-    else if (difference < msPerMonth) {
-      let i = Math.floor(difference/msPerDay);
-      return i + ' day' + isPlural(i) + ' ago';
-    }
-
-    else if (difference < msPerYear) {
-      let i = Math.floor(difference/msPerMonth);
-      return 'approximately ' + i + ' month' + isPlural(i) + ' ago';
-    }
-
-    else {
-      let i = Math.floor(difference/msPerYear);
-      return 'approximately ' + Math.floor(difference/msPerYear ) + ' year' + isPlural(i) + ' ago';
-    }
+  const interval = currentTime - timestamp;
+  const seconds = Math.floor(interval / 1000);
+  const minutes = Math.floor(seconds / 60);
+  const hours = Math.floor(minutes / 60);
+  const days = Math.floor(hours / 24);
+  const months = Math.floor(days / 30);
+  const years = Math.floor(months / 12);
```

Comment:
Moving these `Math.floor` calls and divisions here to the head of the function will have performance implications - in the original code these were in conditional branches, so they were only executed if we reached a branch where it was actually needed. So that's a downside of the new approach, which would matter if this code were in the "hot path" of the application.